lib/ansible/plugins/action/pause.py

```diff
@@ -168,143 +102,67 @@ def run(self, tmp=None, task_vars=None):
         result['start'] = to_text(datetime.datetime.now())
         result['user_input'] = b''
 
-        stdin_fd = None
-        old_settings = None
-        try:
-            if seconds is not None:
-                if seconds < 1:
-                    seconds = 1
+        default_input_complete = None
+        if seconds is not None:
+            if seconds < 1:
+                seconds = 1
 
-                # setup the alarm handler
-                signal.signal(signal.SIGALRM, timeout_handler)
-                signal.alarm(seconds)
+            # show the timer and control prompts
+            display.display("Pausing for %d seconds%s" % (seconds, echo_prompt))
 
-                # show the timer and control prompts
-                display.display("Pausing for %d seconds%s" % (seconds, echo_prompt))
+            # show the prompt specified in the task
+            if new_module_args['prompt']:
                 display.display("(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)\r")
+            else:
+                # corner case where enter does not continue, wait for timeout/interrupt only
+                prompt = "(ctrl+C then 'C' = continue early, ctrl+C then 'A' = abort)\r"
 
-                # show the prompt specified in the task
-                if new_module_args['prompt']:
-                    display.display(prompt)
+            # don't complete on LF/CR; we expect a timeout/interrupt and ignore user input when a pause duration is specified
+            default_input_complete = tuple()
 
-            else:
-                display.display(prompt)
+        # Only echo input if no timeout is specified
+        echo = seconds is None and echo
 
-            # save the attributes on the existing (duped) stdin so
-            # that we can restore them later after we set raw mode
-            stdin_fd = None
-            stdout_fd = None
+        user_input = b''
+        with contextlib.suppress(AnsibleTimeoutExceeded):
             try:
-                stdin = self._connection._new_stdin.buffer
-                stdout = sys.stdout.buffer
-                stdin_fd = stdin.fileno()
-                stdout_fd = stdout.fileno()
-            except (ValueError, AttributeError):
-                # ValueError: someone is using a closed file descriptor as stdin
-                # AttributeError: someone is using a null file descriptor as stdin on windoze
-                stdin = None
-            interactive = is_interactive(stdin_fd)
-            if interactive:
-                # grab actual Ctrl+C sequence
-                try:
-                    intr = termios.tcgetattr(stdin_fd)[6][termios.VINTR]
-                except Exception:
-                    # unsupported/not present, use default
-                    intr = b'\x03'  # value for Ctrl+C
+                if seconds is not None:
+                    # setup the alarm handler
+                    signal.signal(signal.SIGALRM, timeout_handler)
```

Comment:
If we're always remoting to the controller `prompt_until` on the timeout cases anyway and `pause` is already a run-once type action, do we need this `SIGALRM` stuff in the action handler at all anymore?

Reply:
I removed it.